cluster-autoscaler/provisioningrequest/checkcapacity/condition.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package checkcapacity
+
+import (
+	"time"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/apis/autoscaling.x-k8s.io/v1beta1"
+	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/provreqwrapper"
+)
+
+// ProvisioningRequestCondition is a type of Condition that ClusterAutoscaler appends to ProvisioningRequest.
+type ProvisioningRequestCondition string
+
+const (
+
+	// ExpiredCondition is append if the ProvisioningRequest has BookCapacity condition before
+	// and the reservation time is expired or the ProvisioningRequest has Pending condition before
+	// and expiration time is expired.
+	ExpiredCondition = ProvisioningRequestCondition("Expired")
```

Comment:
Also I think we should have those conditions in one place, i.e. in the provreq types package just as the Failed and Capacity Available.